packages/api/cms-api/src/public-upload/public-uploads.controller.ts

```diff
@@ -28,4 +28,17 @@ export class PublicUploadsController {
 
         return publicUploadsFile;
     }
+
+    @Get(":id")
+    @DisableCometGuards()
+    async getFileById(@Param("id") id: string): Promise<PublicUpload> {
+        return this.publicUploadsService.getFileById(id);
+    }
+
+    @Get("download/:id")
+    @DisableCometGuards()
+    async downloadFileById(@Param("id") id: string, @Res() res: NodeJS.WritableStream): Promise<void> {
+        const stream = await this.publicUploadsService.getFileStreamById(id);
+        stream.pipe(res);
```

Comment:
Streaming the file is way more complicated for the DAM than it appears to be here: https://github.com/vivid-planet/comet/blob/main/packages/api/cms-api/src/dam/files/files.controller.ts#L190

@kaufmo @nsams do you know why?

Reply:
this one here doesn't support partial reponse

Reply:
Is this something we need?

Reply:
depends on where it's used.

For PDFs it's used, also for Videos or Downloads (if the Browser supports it and makes use of it) If the uploads are just small profile pictures there is no use for it.

Reply:
It could be used for anything (images, PDFs, videos, ZIP files). 
So should I change this to behave like the DAMs `streamFile()`?

Reply:
I created a follow-up task for this, so we can do this in a separate PR: COM-974
_FYI @thomasdax98_